pkg/controllers/version.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright © 2022 Microshift Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controllers
+
+import (
+	"context"
+
+	"github.com/openshift/microshift/pkg/assets"
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/version"
+	"k8s.io/klog/v2"
+)
+
+type VersionManager struct {
+	cfg *config.MicroshiftConfig
+}
+
+func NewVersionManager(cfg *config.MicroshiftConfig) *VersionManager {
+	s := &VersionManager{}
+	s.cfg = cfg
+	return s
+}
+
+func (s *VersionManager) Name() string { return "version-manager" }
+func (s *VersionManager) Dependencies() []string {
+	return []string{"kube-apiserver", "openshift-apiserver"}
+}
+
+func (s *VersionManager) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+
+	var cm = "assets/version/microshift-version.yaml"
+
+	defer close(stopped)
+	defer close(ready)
+
+	versionInfo := version.Get()
+	var data = map[string]string{
+		"major":   versionInfo.Major,
+		"minor":   versionInfo.Minor,
+		"version": versionInfo.String(),
+	}
+
+	kubeConfigPath := s.cfg.DataDir + "/resources/kubeadmin/kubeconfig"
```

Comment:
filepath.Join?

Reply:
This format was consistent with all of the other path manipulation, but you're right. I've updated this spot and filed #777 to fix it elsewhere in the code base.

Reply:
Thank you, I approved the #777 PR.